cpp/ql/test/library-tests/dataflow/dataflow-tests/test.cpp

```diff
@@ -423,25 +423,32 @@ void intPointerSourceCaller() {
   intPointerSource(&local);
   sink(local); // $ ast,ir=422:7 ast,ir=423:20
 }
-
+// The IR results for this test _is_ equivalent to the AST ones.
+// The IR annotation is just "ir" because the sink of the unitialized source at
+// 428:7 is value of `local`, but the sink of the source from `intPointerSource`
+// value of `*local` (i.e., the indirection node of `local`). So unlike AST dataflow,
+// each of these two sinks correspond to a unique source, and thus we don't need to
+// attach a location annotation to it.
```

Comment:
Should we still provide the location annotations to guarantee that if these get conflated we'll have a test failure?

Reply:
I think that's a good idea. I _think_ this requires adding another tag though, right (since I don't think we can have tags with optional values)?

Reply:
I would prefer to add this in another PR if that's okay with you, though.

Reply:
I think the `428:7` on l. 428 is incorrect now, because this comment was added.

I do have some trouble wrapping my head around the reasoning here. My expectation for `int local[1]` would be that the array is initialised, but the elements of the array are not, but that's apparently not the way to think about this?

Reply:
It looks like the AST library does:
1. Flow from an initialized node to `sink(local)`, _and_ to `sink(*local)`
2. Flow from the elements of the array after leaving `intPointerSource` to `sink(local)`, _and_ to `sink(*local)`.

Reply:
> Correct?

Yes.

> It looks like the AST library does:

I was assuming this is due to conflation problems the AST library has. Or does the AST dataflow library indicate that there may be usability reasons for doing this? I can imagine that it might make sense to consider `int local[1]` to be uninitialised for ease writing some of the sources/sinks that involve arrays.

Reply:
I agree that `UninitializedNode` should probably be handled differently to work for array variables. I think this should be solved in a separate PR, though.